### DISCUSSION TITLE
Expose current user and role to conditional fields

### DISCUSSION THREAD

**Discussion by WoLfulus** [directus/directus#18626] [Feature Requests]
### Summary

I'd like to access to the current user and role inside the conditional settings of a field (instead of just the current item). 

### Basic Example

I have tons of metrics fields in a specific collection, some of them are on the top, some of them are on the bottom of the details page. Some of them are read only and just for quick look. As an admin, I don't care enough to always see them, but I still have a role that the main focus are looking at those metrics but not the other stuff I as an admin look at. Both roles needs to have access to all those metrics, we just don't need them visible all the time.

Another example I have is that I have a role that is able to manage settings of a page, like metadata and other extra stuff, but doesn't actually care about the content itself. For content I have a role for "translators" which manages the page content fields + the translations field. I'd like to hide/show those based on which role is currently viewing the item. Both roles are able to change fields if they need to (from external integration), but doesn't make sense for us to show them inside Directus.

### Motivation

As described on the examples, admins have access to everything by default and there's no way to change that, permissions already solves the problem partially (by hiding fields you don't have access to) but doesn't cover cases where the user/role has access to some field but we just don't want to show it for that role if it doesn't make sense to.

### Detailed Design

When creating conditions to change for example, the visibility of a field, there's no way to create conditions based on who's viewing the item. It's only possible to create conditions based on the current item state only, for example in a collection of "news" posts, if user sets the "highlight" toggle button to true, make a "banner" field show where the user can configure an image to be shown in the hero section of the website.

Together with the item data, the user and/or role should also be allowed to be used as a check, so it allows for tweaking the interface while also keeping permissions of that role/user as it is.

### Must Have

A way to create field conditions based on the current user/role.

### Drawbacks

I can't see any reasons why it "shouldn't" be implemented.

### Alternatives

No alternatives that worked for my use case.

### Adoption Strategy

There's no impact in current state of the application.

### Unresolved Questions

I'm not sure how deep this goes as for implementation cost. I imagine since it's mostly a UI feature, depending on how the settings are stored it's possible to implement without writing a settings migration, but maybe it will require one.

**Comment by BlackDahlia313** [comment]
was just thinking about this where I would love to expose the API for devs in the future, but there are def some items on certain collections I would love to not be exposed.

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by JamesW1** [comment]
Thanks for the suggestion! We've moved feature requests over to our public roadmap at https://roadmap.directus.io/roadmap

I've migrated this request there for you: https://roadmap.directus.io/roadmap?id=735a1b89-eeab-4926-9d43-8ca4cf800ced

Moving forwards, any updates on this subject will be visible on our public roadmap.
